lib/modelMaterialsCommon.js

```diff
@@ -0,0 +1,754 @@
+                                type: WebGLConstants.FLOAT_MAT4
+                            };
+                        }
+                        break;
+                    case 'point':
+                        var point = light.point;
+                        result[lightBaseName + 'Color'] =
+                        {
+                            type: WebGLConstants.FLOAT_VEC3,
+                            value: point.color
+                        };
+                        if (defined(light.node)) {
+                            result[lightBaseName + 'Transform'] =
+                            {
+                                node: light.node,
+                                semantic: 'MODELVIEW',
+                                type: WebGLConstants.FLOAT_MAT4
+                            };
+                        }
+                        result[lightBaseName + 'Attenuation'] =
+                        {
+                            type: WebGLConstants.FLOAT_VEC3,
+                            value: [point.constantAttenuation, point.linearAttenuation, point.quadraticAttenuation]
+                        };
+                        break;
+                    case 'spot':
+                        var spot = light.spot;
+                        result[lightBaseName + 'Color'] =
+                        {
+                            type: WebGLConstants.FLOAT_VEC3,
+                            value: spot.color
+                        };
+                        if (defined(light.node)) {
+                            result[lightBaseName + 'Transform'] =
+                            {
+                                node: light.node,
+                                semantic: 'MODELVIEW',
+                                type: WebGLConstants.FLOAT_MAT4
+                            };
+                            result[lightBaseName + 'InverseTransform'] = {
+                                node: light.node,
+                                semantic: 'MODELVIEWINVERSE',
+                                type: WebGLConstants.FLOAT_MAT4,
+                                useInFragment: true
+                            };
+                        }
+                        result[lightBaseName + 'Attenuation'] =
+                        {
+                            type: WebGLConstants.FLOAT_VEC3,
+                            value: [spot.constantAttenuation, spot.linearAttenuation, spot.quadraticAttenuation]
+                        };
+
+                        result[lightBaseName + 'FallOff'] =
+                        {
+                            type: WebGLConstants.FLOAT_VEC2,
+                            value: [spot.fallOffAngle, spot.fallOffExponent]
+                        };
+                        break;
+                }
+                ++lightCount;
+            }
+        }
+    }
+
+    return result;
+}
+
+function getNextId(dictionary, baseName, startingCount) {
```

Comment:
@leerichard42 are there other parts of gltf-pipeline where we could replace custom code with a call to this function?